src/compiler/debug.ts

```diff
@@ -326,7 +326,17 @@ namespace ts {
             return value.toString();
         }
 
+        const enumMemberCache = new Map<any, SortedReadonlyArray<[number, string]>>();
+
         function getEnumMembers(enumObject: any) {
+            // Assuming enum objects do not change at runtime, we can cache the enum members list
+            // to reuse later. This saves us from reconstructing this each and every time we call
+            // a formatting function (which can be expensive for large enums like SyntaxKind).
+            const existing = enumMemberCache.get(enumObject);

```

Comment:
As you mentioned, this is one of those things that only really matters for tests. In `tsc` there will never be a cached value since these functions should only ever be called right before a crash due to a debug assertion. It can possibly benefit the language service, but I'm not sure how often that would happen in practice.

Reply:
The main benefit at runtime is the debug helpers that we define to use in VS Code's debugger in settings.json, where these helpers are called over and over to show the node types. Some things can be improved later, though.

Reply:
is there any document describing how you debug TypeScript etc? I have my own ways (mostly using `__debugTypeToString` and `__debugGetText` in the debugger) - but I wonder if I'm missing some better techniques.

Reply:
Making sure that `customDescriptionGenerator` is enabled in your `launch.json` is one thing related to this PR, but past that, I don't have any cool tricks besides a collection of helpers I used in the watch window (https://github.com/jakebailey/ts-debug-helpers), which is mainly about printing out nodes/the parser's scanner to more easily see what code is being operated on. Not type related at all.